### PR TITLE
Run Travis on OS X

### PR DESCRIPTION
**Pull request by giordano**
The Clang job on macOS is going to fail because of a couple of errors fixed by #21.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,18 +1,43 @@
...
+  include:
+    - os: linux
+    - os: osx
+      osx_image: xcode10.1
+      env:
+        - MATRIX_EVAL="CC=gcc-7 && CXX=g++-7"
+      addons:
+        homebrew:
+          packages:
+            - gcc@7
```
Comment (sfarrens) [inline .travis.yml:27]: @giordano Is there any reason why you downgraded to `gcc-7` rather than using the current Homebrew default of `gcc-8`?

**Comment by giordano** [comment]
I managed to build Sparse2D also with AppleClang (the default compiler that comes with macOS): https://travis-ci.org/giordano/Sparse2D/builds/506790810  If you're OK I'd cherry-pick [that commit](https://github.com/giordano/Sparse2D/commit/be1a747689b8b2a27f28c603ce25e8d91307010d) into this branch

File: .travis.yml (hunk 1 above)
Comment (giordano) [reply .travis.yml:27]: Not really, I started from a previous Travis script where we were using GCC 7.  I can upgrade to 8 if you want.

Comment (sfarrens) [reply .travis.yml:27]: I think that would be best as most macOS users will use the Homebrew default.

Comment (giordano) [reply .travis.yml:27]: Done!

Comment (giordano) [reply .travis.yml:27]: Well, apparently there was a reason :confused:

Comment (sfarrens) [reply .travis.yml:27]: Odd, it builds locally with `gcc-8` without any problems. You can revert to `gcc-7` as it seems to work on Travis.

Comment (giordano) [reply .travis.yml:27]: That might be a Travis quirk

**Pull request merged**